src/mem/sizeclasstable.h

```diff
@@ -5,31 +5,49 @@
 
 namespace snmalloc
 {
+  constexpr size_t PTR_BITS = bits::next_pow2_bits_const(sizeof(void*));
+
+  constexpr static size_t sizeclass_lookup_index(size_t s)
+  {
+    // We shift by PTR_BITS as makes code-gen for the table lookup nicer.
+    // We could shift by MIN_ALLOC_BITS, but then there is a slightly more
+    // complex sequence.
+    return (s - 1) >> PTR_BITS;
```

Comment:
I think I was more confused after reading this comment than before.  I guess the key point is that we can index the table by anything we like that's cheap to compute, and this particular shift makes it cheap, at the expense of a larger table?